appcues/src/main/java/com/appcues/debugger/screencapture/AndroidTargetingStrategy.kt

```diff
@@ -170,12 +171,26 @@ private fun View.asCaptureView(screenBounds: Rect): ViewElement? {
             displayName = it?.displayName,
             selector = it,
             type = it?.type ?: this::class.java.simpleName,
-            children = if (children.isEmpty()) null else children,
+            children = children.ifEmpty { null },
         )
     }
 }
 
-private fun View.selector(): AndroidViewSelector? {
+private fun View.isFullyVisible(globalVisibleRect: Rect, tolerance: Double = 0.1): Boolean {
+    val visibleHeight = globalVisibleRect.height().toDouble()
+    val actualHeight = measuredHeight.toDouble()
+    val visibleWidth = globalVisibleRect.width().toDouble()
+    val actualWidth = measuredWidth.toDouble()
+
+    val heightVisible = visibleHeight > actualHeight * (1.0 - tolerance)
+    val widthVisible = visibleWidth > actualWidth * (1.0 - tolerance)
+    return heightVisible && widthVisible
+}
+
+private fun View.selector(globalVisibleRect: Rect): AndroidViewSelector? {
+    // if the entire view is not visible (within tolerance) exclude selector for targeting
+    if (isFullyVisible(globalVisibleRect).not()) return null
```

Comment:
this new check ensures that 90% of the height and width of a view will be visible on screen before any tooltip can point to it - to handle things scrolling into the view.

the Compose side didn't really need such a check (i think) as we are already constructing the view rect and checking if it is within screen bounds (new check on L218) - need to test/confirm behavior on this part.